dlgr/griduniverse/static/scripts/demo.js

```diff
@@ -353,31 +354,47 @@ var playerSet = (function () {
       return Object.keys(this._players).length;
     };
 
-    PlayerSet.prototype.update = function (playerData) {
-      var currentPlayerData,
-          oldPlayerData,
+    PlayerSet.prototype.update = function (allPlayersData) {
+      var freshPlayerData,
+          existingPlayer,
           i;
 
-      for (i = 0; i < playerData.length; i++) {
-        currentPlayerData = playerData[i];
-        oldPlayerData = this._players[currentPlayerData.id];
-        if (oldPlayerData && oldPlayerData.id === this.ego_id) {
+      for (i = 0; i < allPlayersData.length; i++) {
+        freshPlayerData = allPlayersData[i];
+        existingPlayer = this._players[freshPlayerData.id];
+        if (existingPlayer && existingPlayer.id === this.ego_id) {
+
           /* Don't override current player motion timestamp */
-          currentPlayerData.motion_timestamp = oldPlayerData.motion_timestamp;
-          /* Only override position from server if tremble is enabled,
-             otherwise motion jitter is likely and positions will sync anyway. */
-          if (settings.motion_tremble_rate === 0) {
-            currentPlayerData.position = oldPlayerData.position;
+          freshPlayerData.motion_timestamp = existingPlayer.motion_timestamp;
+
+          // Only override position from server if tremble is enabled,
+          // or if we know the Player's position is out of sync with the server.
+          // Otherwise, the ego player's motion is constantly jittery.
+          if (settings.motion_tremble_rate === 0 && existingPlayer.positionInSync) {
+            freshPlayerData.position = existingPlayer.position;
+          } else {
+            console.log("Overriding position from server!");
           }
         }
         var last_dimness = 1;
-        if (this._players[currentPlayerData.id] !== undefined) {
-          last_dimness = this._players[currentPlayerData.id].dimness;
-        } 
-        this._players[currentPlayerData.id] = new Player(currentPlayerData, last_dimness);
+        if (this._players[freshPlayerData.id] !== undefined) {
+          last_dimness = this._players[freshPlayerData.id].dimness;
+        }
+        this._players[freshPlayerData.id] = new Player(freshPlayerData, last_dimness);
       }
     };
 
+    PlayerSet.prototype.startScheduledAutosyncOfEgoPosition = function () {
```

Comment:
How much benefit does this provide? It seems like the `move_rejection` should handle most or all sync issues that wouldn't recover on their own. Perhaps we should only do this if we detect a delta position larger than `n` squares? That would require keeping track of the most recent delta, but that shouldn't be too hard. This seems like it could result in additional issues for players moving relatively fast.

Reply:
The harder it became to get out of sync, the more frequently I found that once I was out of sync, I'd *stay* out if sync, since it could be quite a while before another `IllegalMove` would be raised. I'd go bashing into other players to try to trigger something to get back in sync. So, I do think it's a significant part of the solution, unfortunately.

Reply:
@alecpm I did miss pulling in a commit here. I'd removed the logging call that's now in `parse_message()`:
```
    def parse_message(self, raw_message):
        if raw_message.startswith(self.channel + ":"):
            logger.info(
                "We received a message for our channel: {}".format(raw_message)
            )
...
```
Interesting that your experience was still pretty good despite that being left in. How do you feel about removing it? Related to this, I'd also been wondering about output buffering (or lack thereof). Maybe we investigate that first? I hate the idea of logging modifying behavior, since that's a recipe for Heisenbugs.

Reply:
That seems like a deeply unnecessary log, and if we're logging such things we should log at debug level or lower. I'm fine leaving it out, but I think looking into our default logging config is probably a good longer term idea. We have a `logging.ini` in dallinger that's used for the commandline, but I don't think that's used on the Heroku side. It looks like `MemoryHandler` is intended for this sort of thing.

Reply:
logging overhead gave me no problems on re-test, so I'm happy to defer that whole question for now.

Reply:
I nuked it, since it's almost always too verbose to be useful.